server/tserver/src/main/java/org/apache/accumulo/tserver/compaction/EverythingCompactionStrategy.java

```diff
@@ -27,7 +27,7 @@
 
   @Override
   public boolean shouldCompact(MajorCompactionRequest request) throws IOException {
-    return request.getFiles().size() > 0;
+      return true; // ACCUMULO-3645 compact for empty files too
```

Comment:
I think I'd only be in favor of making the change for 1.6.3 if it's done as a new CompactionStrategy. I'd need convincing to change the default implementation in a bug fix release without seriously enumerating the corner cases.